### PR TITLE
Move SemVerHandler to analysis to parsing to increase modularity...

### PR DESCRIPTION
**Pull request by rocky**
If ever you want to separate building analysis from scanning/parsing/ast, this will help.

### Checklist
- [x] Code compiles correctly
- [x] All tests are passing
- [ ] New tests have been created which fail without the change (if possible)
- [ ] README / documentation was extended, if necessary
- [ ] Changelog entry (if change is visible to the user)
- [x] Used meaningful commit messages

### CODE REVIEW

**Review by chriseth — CHANGES_REQUESTED** [review]
Since the SyntaxChecker also relies on the SemVerHandler, I think it would be best to move it all the way down to liblangutil. SemVerChecker.h includes `parsing/Token.h` but that is actually an "alias" for `liblangutil/Token.h`.

**Comment by rocky** [comment]
Ah - okay. Thanks for the information. Will make the changes later today.

**Comment by stackenbotten** [comment]
There was an error when running `test_check_style` for commit `389a56be693d324fdeddb2ec36ad12cb93e0b94a`:
```
Coding style error:
 liblangutil/SemVerHandler.h:36: class SemVerError: dev::Exception

```
Please check that your changes are working as intended.

**Comment by chriseth** [comment]
Please squash into one commit.

**Comment by stackenbotten** [comment]
There was an error when running `test_check_style` for commit `ebe6413f39f62e3a43711ff29461fe0411b4a8b1`:
```
Error: Trailing whitespace found:
 liblangutil/ParserBase.cpp.rej:6: 

```
Please check that your changes are working as intended.

**Comment by rocky** [comment]
@chriseth ok I think everything is squashed to one commit now.

File: liblangutil/SemVerHandler.h [hunk 1]
```diff
@@ -31,6 +31,8 @@ namespace dev
 namespace solidity
```
Comment (chriseth) [inline liblangutil/SemVerHandler.h]: The namespace should change now, too.

Comment (rocky) [reply liblangutil/SemVerHandler.h]: should be hanled in e0de5aa - but please double check.

File: liblangutil/SemVerHandler.h [hunk 2]
```diff
@@ -22,15 +22,15 @@
...
 #include <string>
 #include <vector>
 
-namespace dev
-{
-namespace solidity
+namespace langutil
 {
 
+using ::langutil::Token;
```
Comment (chriseth) [inline liblangutil/SemVerHandler.h]: This should have no effect now and should be deleted.

Comment (rocky) [reply liblangutil/SemVerHandler.h]: Ok - how about now in b42446 ?

**Comment by rocky** [comment]
@chriseth BTW after this, the next up on my list of Modularity things is moving `Version.h` from `libsolsolidity/interface` to some place eariler, like `libsolidity/parsing`, or if you can suggest some other place, that would be good.

**Review by chriseth — APPROVED** [review]

**Pull request merged**